### PR TITLE
Fixes for Building with Update 1

### PR DESCRIPTION
**Pull request by jasonmalinowski**
A set of changes to update the Roslyn solution to build with Update 1. These changes are needed since we've formalized how project.json works with desktop-targeting clas libraries. The changes should all be backwards compatible with the original release of Visual Studio 2015 and should be an effective no-op for anybody building there.

These changes are targeting stabilization because once Update 1 is released, our community will be branching off of stabilization (although submitting PRs to master) to ensure they're working on the same base and won't have internal API breaks.

Reviewers: @dotnet/roslyn-ide, @kevinpi, @tannergooding, @agocke

### CODE REVIEW

**Comment by davkean** [comment]
:+1: The changes look good, though:

* _Remove broken references to old-style NuGet references_ - already in as https://github.com/dotnet/roslyn/commit/ed305aa706f7b575429b14455b488d8ee4af5a22.
* _Normalize the Roslyn.sln project GUID casing_ - under review as https://github.com/dotnet/roslyn/pull/6203

**Comment by davkean** [comment]
You already said the same thing to me - damn, no longer getting notifications anymore.

**Comment by jasonmalinowski** [comment]
retest Linux please

**Comment by jasonmalinowski** [comment]
retest linux please

**Comment by jasonmalinowski** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by jasonmalinowski** [comment]
Retest this please. Having to re-run since commits were added that Jenkins didn't pick up.

File: src/Compilers/CSharp/csc/project.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
     "net45": { }
   },
   "runtimes": {
+    "win7": { },
     "win7-anycpu": { },
```
Comment (Pilchie) [inline src/Compilers/CSharp/csc/project.json]: Should we be removing these, or are you leaving them for compat with VS2015 RTM?

**Comment by Pilchie** [comment]
:+1:

(Aside: I wonder if we should ask GitHub to suppress diffs in project.lock.json files by default, the way they do in resx designer generated files?)

**Comment by Pilchie** [comment]
Tagging @yishaigalatzer and @davidfowl for the question above.

**Comment by yishaigalatzer** [comment]
I thought the main reason you checked the lock files in is to see the diffs?

**Comment by amcasey** [comment]
FYI failure is https://github.com/dotnet/roslyn/issues/6190.  @davkean is working on a mitigation.

**Comment by davkean** [comment]
retest prtest/win/dbg/unit64 please

**Comment by Pilchie** [comment]
@yishaigalatzer we want to know if a lock file changed when we didn't expect it to, but not necessarily to see every diff when we are mucking with project.json files.

File: src/Compilers/CSharp/csc/project.json (hunk 1 above)
Comment (jasonmalinowski) [inline src/Compilers/CSharp/csc/project.json]: I'm leaving them in for compat with RTM until we declare RTM no longer suitable for Roslyn development. More practically, our build machines still need that!

**Comment by jasonmalinowski** [comment]
@yishaigalatzer: and it's also a workaround for restore instability on some of the non-Windows platforms we're still bringing up.

**Comment by jasonmalinowski** [comment]
The re-push was just to rebase this atop the latest stabilization, since there's been a few days of changes I want to ensure I cleanly test with.

**Comment by diryboy** [comment]
Isn't branching off stabilization but target master in pr still merge and pollute master breaking RTM? Did you mean need rebase to master before submitting?

BTW does VS Update1's git support rebasing?

**Comment by jasonmalinowski** [comment]
@diryboy: the expectation (at least for people doing IDE bug fixing) will be to branch off of stabilization, since that ensures that you have internal APIs that are synced with VS synced. But you'll submit to master. This won't pollute master since we're going to be merging stabilization back down into master any time we make a change in stabilization before Update 1 ships.

**Comment by jasonmalinowski** [comment]
If you're making changes in the compiler, you might do other workflows.

**Pull request merged**